lib/ecto/adapters/mysql/connection.ex

```diff
@@ -769,20 +779,23 @@ if Code.ensure_loaded?(Mariaex) do
     defp ecto_cast_to_db(type, query), do: ecto_to_db(type, query)
 
     defp ecto_to_db(type, query \\ nil)
-    defp ecto_to_db({:array, _}, query),      do: error!(query, "Array type is not supported by MySQL")
-    defp ecto_to_db(:id, _query),             do: "integer"
-    defp ecto_to_db(:serial, _query),         do: "bigint unsigned not null auto_increment"
-    defp ecto_to_db(:bigserial, _query),      do: "bigint unsigned not null auto_increment"
-    defp ecto_to_db(:binary_id, _query),      do: "binary(16)"
-    defp ecto_to_db(:string, _query),         do: "varchar"
-    defp ecto_to_db(:float, _query),          do: "double"
-    defp ecto_to_db(:binary, _query),         do: "blob"
-    defp ecto_to_db(:uuid, _query),           do: "binary(16)" # MySQL does not support uuid
-    defp ecto_to_db(:map, _query),            do: "text"
-    defp ecto_to_db({:map, _}, _query),       do: "text"
-    defp ecto_to_db(:utc_datetime, _query),   do: "datetime"
-    defp ecto_to_db(:naive_datetime, _query), do: "datetime"
-    defp ecto_to_db(other, _query),           do: Atom.to_string(other)
+    defp ecto_to_db({:array, _}, query),           do: error!(query, "Array type is not supported by MySQL")
+    defp ecto_to_db(:id, _query),                  do: "integer"
+    defp ecto_to_db(:serial, _query),              do: "bigint unsigned not null auto_increment"
+    defp ecto_to_db(:bigserial, _query),           do: "bigint unsigned not null auto_increment"
+    defp ecto_to_db(:binary_id, _query),           do: "binary(16)"
+    defp ecto_to_db(:string, _query),              do: "varchar"
+    defp ecto_to_db(:float, _query),               do: "double"
+    defp ecto_to_db(:binary, _query),              do: "blob"
+    defp ecto_to_db(:uuid, _query),                do: "binary(16)" # MySQL does not support uuid
+    defp ecto_to_db(:map, _query),                 do: "text"
+    defp ecto_to_db({:map, _}, _query),            do: "text"
+    defp ecto_to_db(:time_usec, _query),           do: "time"
+    defp ecto_to_db(:utc_datetime, _query),        do: "datetime"
+    defp ecto_to_db(:utc_datetime_usec, _query),   do: "datetime"
+    defp ecto_to_db(:naive_datetime, _query),      do: "datetime"
+    defp ecto_to_db(:naive_datetime_usec, _query), do: "datetime"
```

Comment:
If we return `datetime(6)` here, do we run into issues?

Reply:
Having this just be `"timestamp"` here allows the precision option to be set. For mysql, a default precision of `6` is set for `_usec` types in `column_type/2`.

Reply:
You are correct, thank you!